users/ignoring.rst

```diff
@@ -88,6 +88,32 @@ The ``.stignore`` file contains a list of file or path patterns. The
    are *included* (that is, *not* ignored). This can be used to override
    more general patterns that follow.
 
+   .. note::
+
+      Negated patterns that can match items below the folder root will cause
+      Syncthing to traverse otherwise ignored directories. If the
+      :ref:`watcher <scanning>` is enabled, those directories will also be
+      watched. Directories ignored before the first negated pattern can
+      however be safely skipped, since the first matching pattern wins. For
+      example::
+
+         /foo
+         /bar
+         !baz
+         *
+
+      The directories ``foo`` and ``bar`` will be entirely ignored. However any
+      other directories present must be scanned entirely to find any items
+      named `baz`, despite the fact that they will be ignored due to the
+      ``*``. As a special case, top-level rooted patterns (e.g. ``!/foo``) do
+      not cause this behaviour::
+
+         !/baz
+         *
+
+      In this case, only the directory ``baz`` will be scanned, since
+      everything else is ignored by the ``*`` pattern.
```

Comment:
Despite having seen (but not thoroughly checked) the related PR, this special case suprised me.  I.e. could be better explained.